Formula/lua@5.2.rb

```diff
@@ -0,0 +1,205 @@
+    end
+
+    # Fix path in the config header
+    inreplace "src/luaconf.h", "/usr/local", HOMEBREW_PREFIX
+
+    # We ship our own pkg-config file as Lua no longer provide them upstream.
+    system "make", "macosx", "INSTALL_TOP=#{prefix}", "INSTALL_MAN=#{man1}", "INSTALL_INC=#{include}/lua-5.2"
+    system "make", "install", "INSTALL_TOP=#{prefix}", "INSTALL_MAN=#{man1}", "INSTALL_INC=#{include}/lua-5.2"
+    (lib/"pkgconfig/lua-5.2.pc").write pc_file
+
+    # Renaming from Lua to Lua52.
+    # Note that the naming must be both lua-version & lua.version.
+    # Software can't find the libraries without supporting both the hyphen or full stop.
+    mv "#{bin}/lua", "#{bin}/lua-5.2"
+    mv "#{bin}/luac", "#{bin}/luac-5.2"
+    mv "#{man1}/lua.1", "#{man1}/lua-5.2.4"
+    mv "#{man1}/luac.1", "#{man1}/luac-5.2.4"
+    bin.install_symlink "lua-5.2" => "lua5.2"
+    bin.install_symlink "luac-5.2" => "luac5.2"
+    include.install_symlink "lua-5.2" => "lua5.2"
+    (lib/"pkgconfig").install_symlink "lua-5.2.pc" => "lua5.2.pc"
+    (libexec/"lib/pkgconfig").install_symlink lib/"pkgconfig/lua-5.2.pc" => "lua.pc"
+
+    # This resource must be handled after the main install, since there's a lua dep.
+    # Keeping it in install rather than postinstall means we can bottle.
+    if build.with? "luarocks"
+      resource("luarocks").stage do
+        ENV.prepend_path "PATH", bin
+
+        system "./configure", "--prefix=#{libexec}", "--rocks-tree=#{HOMEBREW_PREFIX}",
+                              "--sysconfdir=#{etc}/luarocks52", "--with-lua=#{prefix}",
+                              "--lua-version=5.2", "--versioned-rocks-dir"
+        system "make", "build"
+        system "make", "install"
+
+        (pkgshare/"5.2/luarocks").install_symlink Dir["#{libexec}/share/lua/5.2/luarocks/*"]
+        bin.install_symlink libexec/"bin/luarocks-5.2"
+        bin.install_symlink libexec/"bin/luarocks-admin-5.2"
+
+        # This block ensures luarock exec scripts don't break across updates.
+        inreplace libexec/"share/lua/5.2/luarocks/site_config.lua" do |s|
+          s.gsub! libexec.to_s, opt_libexec
+          s.gsub! include.to_s, "#{HOMEBREW_PREFIX}/include"
+          s.gsub! lib.to_s, "#{HOMEBREW_PREFIX}/lib"
+          s.gsub! bin.to_s, "#{HOMEBREW_PREFIX}/bin"
+        end
+      end
+    end
+  end
+
+  def pc_file; <<~EOS
+    V= 5.2
+    R= 5.2.4
+    prefix=#{opt_prefix}
+    INSTALL_BIN= ${prefix}/bin
+    INSTALL_INC= ${prefix}/include/lua-5.2
+    INSTALL_LIB= ${prefix}/lib
+    INSTALL_MAN= ${prefix}/share/man/man1
+    INSTALL_LMOD= #{HOMEBREW_PREFIX}/share/lua/${V}
+    INSTALL_CMOD= #{HOMEBREW_PREFIX}/lib/lua/${V}
+    exec_prefix=${prefix}
+    libdir=${exec_prefix}/lib
+    includedir=${prefix}/include/lua-5.2
+
+    Name: Lua
+    Description: An Extensible Extension Language
+    Version: #{version}
+    Requires:
+    Libs: -L${libdir} -llua.5.2 -lm
+    Cflags: -I${includedir}
+    EOS
+  end
+
+  def caveats; <<~EOS
```

Comment:
Should probably decide what to do with these caveats and then harmomise it across all the Lua versions.